firedrake/preconditioners/asm.py

```diff
@@ -36,10 +36,24 @@ def initialize(self, pc):
         # Create new PC object as ASM type and set index sets for patches
         asmpc = PETSc.PC().create(comm=pc.comm)
         asmpc.incrementTabLevel(1, parent=pc)
-        asmpc.setOptionsPrefix(self.prefix + "_sub")
+        asmpc.setOptionsPrefix(self.prefix + "sub_")
```

Comment:
I _think_ we want the asmpc to have prefix `self.prefix` (see below). But am willing to be persuaded otherwise. The way I figure it is that we configure this python pc with the options prefix from `pc.getOptionsPrefix()` and then the concatenation of that with `self._prefix` becomes the options prefix for the inner thing.

But maybe that doesn't quite work out if we want to do options lookup to configure the python thing. Hmmm....

Reply:
Let us do this in another round consistently for all of the PCs.